### PR TITLE
Fixed activity indicator that was always visible and not animating

### PR DESCRIPTION
**Pull request by lokeshub**
On Add Location screen, now activity indicator is stopped and hidden when query not running and animates when querying.

### CODE REVIEW

File: WeatherInfo/ViewControllers/Locations View Controller/Add Location View Controller/AddLocationViewController.swift [hunk 1]
```diff
@@ -38,19 +38,21 @@ class AddLocationViewController: UIViewController {
         viewModel.locationsDidChange = { [unowned self] (locations) in
             self.tableView.reloadData()
         }
-        
+        self.activityIndicatorView.isHidden = true
```
Comment (alokc83) [inline WeatherInfo/ViewControllers/Locations View Controller/Add Location View Controller/AddLocationViewController.swift]: Why this is needed when you are changing states in ViewDidAppear?

Comment (lokeshub) [reply WeatherInfo/ViewControllers/Locations View Controller/Add Location View Controller/AddLocationViewController.swift]: When I ran in slow animation it is visible there unanimated when screen is loaded and disappears when the first time search is made.

Comment (lokeshub) [inline WeatherInfo/ViewControllers/Locations View Controller/Add Location View Controller/AddLocationViewController.swift]: On main story board activity indicator's property, selected "Hide when Stopped'. No need to manually hide in code after this.

File: WeatherInfo/ViewControllers/Locations View Controller/Add Location View Controller/AddLocationViewController.swift [hunk 2]
```diff
@@ -38,19 +38,21 @@ class AddLocationViewController: UIViewController {
...
+                self.activityIndicatorView.isHidden = false
                 self.activityIndicatorView.startAnimating()
             } else {
+                self.activityIndicatorView.isHidden = true
                 self.activityIndicatorView.stopAnimating()
             }
         }
-    }
-
-    override func viewDidAppear(_ animated: Bool) {
```
Comment (alokc83) [inline WeatherInfo/ViewControllers/Locations View Controller/Add Location View Controller/AddLocationViewController.swift:51]: Why did you remove viewDidAppear

**Review by alokc83 — CHANGES_REQUESTED** [review]

File: WeatherInfo/ViewControllers/Locations View Controller/Add Location View Controller/AddLocationViewController.swift (hunk 2 above)
Comment (alokc83) [reply WeatherInfo/ViewControllers/Locations View Controller/Add Location View Controller/AddLocationViewController.swift:51]: I would want searchBar to become the first responder when the view is being presented.

**Comment by alokc83** [comment]
What have you done to fix the issue?

File: WeatherInfo/ViewControllers/Locations View Controller/Add Location View Controller/AddLocationViewController.swift (hunk 2 above)
Comment (lokeshub) [reply WeatherInfo/ViewControllers/Locations View Controller/Add Location View Controller/AddLocationViewController.swift:51]: Yes it is searchBar is made first responder still in viewDidAppear. I did not remove it.

**Comment by lokeshub** [comment]
ActivityIndicator's property in storyboard to hide when not animating was enabled.
And this status change was not required in code after that

File: WeatherInfo/ViewControllers/Locations View Controller/Add Location View Controller/AddLocationViewController.swift (hunk 1 above)
Comment (alokc83) [reply WeatherInfo/ViewControllers/Locations View Controller/Add Location View Controller/AddLocationViewController.swift]: What about the bug that you mentioned yesterday?

Comment (lokeshub) [reply WeatherInfo/ViewControllers/Locations View Controller/Add Location View Controller/AddLocationViewController.swift]: This was introducing that bug. The activity indicator was already not hidden when the view loaded. Once we set this above property, we have to just start/Stop the animation to make the activity indicator view they way it should work.

Comment (alokc83) [reply WeatherInfo/ViewControllers/Locations View Controller/Add Location View Controller/AddLocationViewController.swift]: Then hidden is not the way. you should or should not be animating. 
As property says hide when stop animating. 
@lokeshub

**Comment by alokc83** [comment]
Read my comment @lokeshub

**Comment by lokeshub** [comment]
Closing this as invalid changes. I will work on fresh PR.
